srcs/Server.cpp

```diff
@@ -2,117 +2,149 @@
-	this->sa = src.sa;
 }
 
 Server&	Server::operator=(const Server& rhs) {
 	this->listenSocket = rhs.listenSocket;
 	this->currSocket = rhs.currSocket;
 	this->hints = rhs.hints;
 	this->servinfo = rhs.servinfo;
-	this->sa = rhs.sa;
 	return (*this);
 }
 
-// HELPER FUNCTIONS
-void	sigchld_handler(int s)
-{
-	(void)s; // WHY???????????????????????????????????
-	// waitpid() might overwrite errno, so we save it and restore it:
-	//int saved_errno = errno;
-
-	while (waitpid(-1, NULL, WNOHANG) > 0);
-	//errno = saved_errno;
-}
-
-// METHODS
+// PUBLIC METHODS
 void	Server::run(void) {
-	setup();
-}
+	int poll_count;
 
-void	Server::setup(void) {
-	int				err_val;
-	struct addrinfo *p;
-	char	recv_buff[BUFF_SIZE]; // 1 kiloByte
-
-	// Fill the hints struct with information for possible connections
-	memset(&hints, 0, sizeof(hints));
-	hints.ai_family = AF_INET;			// Accept IPv4 connections
-	hints.ai_socktype = SOCK_STREAM;	// Accept over TCP (not UDP)
-	hints.ai_flags = AI_PASSIVE;		// Assign the address of my local host to the socket structures.
-	if ((err_val = getaddrinfo(NULL, PORT, &hints, &servinfo)) != 0)
-	{
-		std::cerr << "error: getaddrinfo: " << gai_strerror(err_val) << std::endl;
-		exit(-1); // CHANGE THISS!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-	}
-	std::cout << servinfo->ai_canonname << std::endl;
-	// If everything worked, servinfo is a linked list with sockaddr we can use
-	int count = -1;
-	for (p = servinfo; p != NULL; p = p->ai_next)
+	std::cout << "Launching server..." << std::endl;
+	while (1)
 	{
-		count++;
-		try {
-			listenSocket.createSocket(p->ai_family, p->ai_socktype, p->ai_protocol);
-		} catch (std::exception& e) {
-			std::cerr << "struct addrinfo #"<< count << ": " << e.what() << std::endl;
-			continue ;
+		if ((poll_count = poll(pfds, fd_count, -1)) < 0)
+		{
+			std::cout << "heh" << std::endl;
+			throw PollException();
 		}
-		try {
-			listenSocket.bindSocket(p->ai_addr, p->ai_addrlen);
-		} catch (std::exception& e) {
-			std::cerr << "struct addrinfo #"<< count << ": " << e.what() << std::endl;
-			std::cout << p->ai_next << std::endl;
-			continue ;
+		// Run through the existing connections looking for data to read
+		for (int i = 0; i < fd_count; i++)
+		{
+			// Check if descriptor has data available
+			if (pfds[i].revents & POLLIN)
+			{
+				if (pfds[i].fd == listenSocket.getSocketFD())
+					handleConnection();
+				else
+					handleRequest(i);
+			}
 		}
-		break ;
 	}
-	// When the loop is complete, we should have a socket bound to our address, we can free now
-	freeaddrinfo(servinfo);
-	if (p == NULL)
-		throw Socket::SocketBindException();
-	listenSocket.listenOnSocket();
-
-	sa.sa_handler = sigchld_handler;
-	sigemptyset(&sa.sa_mask);
-	sa.sa_flags = SA_RESTART;
-	if (sigaction(SIGCHLD, &sa, NULL) < 0)
+}
+
+// PRIVATE METHODS
+void	Server::handleRequest(int i) {
+	int	nbytes;
+	int sender_fd;
+	char buf[1024];    // Buffer for client data
+
+	sender_fd = pfds[i].fd;
+	memset(buf, 0, 1024);
```

Comment:
same here for what it looks like BUFFERSIZE

Reply:
Done!